frontend/src/pages/device/Snapshots/index.vue

```diff
@@ -334,6 +342,26 @@ export default {
                 console.error(err)
                 Alerts.emit('Failed to get snapshot.', 'warning')
             })
+        },
+        showDownloadSnapshotDialog (snapshot) {
+            this.$refs.snapshotExportDialog.show(snapshot)
+        },
+        showDeploySnapshotDialog (snapshot) {
+            this.deploySnapshot(snapshot.id)
+        },
+
+        // enable/disable snapshot actions
+        canDeploy (_row) {
```

Comment:
This methods would be better suite d in computed props which would also sync the state of the button automatically when rows would change. Using methods to determine state changes could cause unexpected behavior (elements not updating after state changes) in the ui

Reply:
Discussed & I understand the better way now but not in scope at this time

Once Serban has implemented the first one and we establish a pattern, these will be retro'd upon working in the file